core/src/main/scala/org/typelevel/ci/CIString.scala

```diff
@@ -65,6 +65,9 @@ final class CIString private (override val toString: String)
 @suppressUnusedImportWarningForCompat
 object CIString {
   def apply(value: String): CIString = new CIString(value)
+  def transform(f: String => String): CIString = CIString(f(toString))
```

Comment:
I think you're transforming the `toString` of the companion object here.  This belongs on the class.